CorsixTH/Src/th_gfx_sdl.cpp

```diff
@@ -119,6 +119,33 @@ void getEnclosingScaleRect(const SDL_Rect* rect, double scale_factor,
   dst_rect->y = dst_y;
 }
 
+#if SDL_VERSION_ATLEAST(2, 0, 10)
+
+// If using SDL 2.0.10 or newer, use SDL_FRect to get better precision on scaled
+// rendering.
+void getScaleRect(const SDL_Rect* rect, double scale_factor,
+                  SDL_FRect* dst_rect) {
+  dst_rect->x = static_cast<float>(rect->x * scale_factor);
+  dst_rect->y = static_cast<float>(rect->y * scale_factor);
+  dst_rect->w = static_cast<float>(rect->w * scale_factor);
+  dst_rect->h = static_cast<float>(rect->h * scale_factor);
+}
+
+#else
+
+// Prior to SDL 2.0.10, fallback to using the enclosing integer SDL_Rect for
+// scaled rendering.
+#define SDL_FRect SDL_Rect
+#define SDL_RenderCopyF SDL_RenderCopy
+#define SDL_RenderCopyExF SDL_RenderCopyEx
```

Comment:
Not sure if you'd prefer we make up names e.g. SDL_SCALED_RECT, SDL_RENDER_COPY_SCALED, SDL_RENDER_COPY_SCALED_EX to avoid the presumption that where you see SDL_FRect you always have a floating point rect rather than it depending on sdl version. Let me know I'm happy to change it.

Reply:
This is fine. I doubt we'll be doing much more work with older SDL versions.